subprojects/diagnostics/src/integTest/groovy/org/gradle/api/tasks/diagnostics/DependencyReportTaskIntegrationTest.groovy

```diff
@@ -900,4 +902,122 @@ conf
 
 """
     }
+
+    @Unroll
+    def "renders custom dependency constraint reasons"() {
```

Comment:
I'm not sure this is a good idea. I'd leave it so that reasons are only displayed in the dependency insight report.

Reply:
Why so? In my mind the dependency report is lacking behind, and it's unfortunate that it doesn't show the reasons. Without this, you'd have to execute the insight report multiple times for each dependency to understand if one comes from, say, a platform definition, or a transitive dependency.

Reply:
Two reasons:
1) I think build scans can do a much better job of rendering this information
2) The `dependencies` report is nice and easy to scan right now. Adding a lot of text might make it less usable. _might_

Reply:
Ok I'll revert the change for now.